algorithm/cgal/repair_util.h

```diff
@@ -157,3 +181,98 @@ static bool repair_self_intersections(Surface_mesh& mesh) {
       mesh, {REPAIR_AUTOREFINE_AND_REMOVE_SELF_INTERSECTIONS,
              REPAIR_TRY_REMOVE_SELF_INTERSECTIONS});
 }
+
+template <typename K>
+static bool repair_degeneracies(CGAL::Surface_mesh<typename K::Point_3>& mesh) {
+  return CGAL::Polygon_mesh_processing::remove_degenerate_edges(mesh) &&
+         CGAL::Polygon_mesh_processing::remove_degenerate_faces(mesh);
+}
+
+template <typename K>
+static bool repair_manifold(CGAL::Surface_mesh<typename K::Point_3>& mesh) {
+  typedef CGAL::Surface_mesh<typename K::Point_3> Surface_mesh;
+  // This repairs cases where we have two corners that touch at one coordinate.
+
+  // Duplicating the vertices fixes the topological problem, but not the
```

Comment:
There are a lot of corner cases to handle 😉

Reply:
Yeah, hopefully some are redundant or can be merged.